core/contractsdk/cpp/xchain/builtins.cc

```diff
@@ -0,0 +1,20 @@
+#include "xchain/builtins.h"
+#include <string>
+
+extern "C" void xvm_hash(const char* name, const char* inputptr, int inputlen,
+                         char* outputptr, int outputlen, int hex_encode);
+
+namespace xchain {
+std::string sha256(const std::string& input) {
```

Comment:
hash/crypto functions could be aggregated in a sub namespace?

Reply:
Makes sense, has been added.